tests/general/QuotesBearTest.py

```diff
@@ -97,3 +97,19 @@ def test_invalid_quotes(self):
                              '-"a string with double quotes!"\n'
                              "+'a string with double quotes!'\n"
                              " 'A single quoted string with \" in it'\n")
+
+        self.section['preferred_quotation'] = '"'
+        with execute_bear(self.uut, self.filename, self.double_quote_file,
+                          dependency_results=self.dep_results,
+                          force_preferred_quotation=True) as results:
+            res_list = list(results)
+            self.assertEqual(len(res_list), 1)
+            self.assertEqual(res_list[0].diffs[self.filename].unified_diff,
+                             '--- \n'
+                             '+++ \n'
+                             '@@ -3,4 +3,4 @@\n'
+                             ' Multiline string\n'
+                             " '''\n"
+                             ' "a string with double quotes!"\n'
+                             "-'A single quoted string with \" in it'\n"
+                             '+"A single quoted string with \" in it"\n')
```

Comment:
Shouldn't it be `\\"`, because the escape character has to appear in the patch?

Reply:
no, why the escape character needs to appear in the patch?

Reply:
no need right? @Makman2

Reply:
Let's consider the test file:
```
'''
Multiline string
'''
"a string with double quotes!"
'A single quoted string with " in it'
```
Because we force quotation with `"`, the last string has to be changed because it uses `'` quotes (let's not care for `'''` right now):
```
'A single quoted string with " in it'
```
The single quotes need to be replaced with double quotes:
```
"A single quoted string with " in it"
```
The problem: The double quote in the middle. It's part of the string, so it has to be escaped:
```
"A single quoted string with \" in it"
```
Because you are entering the string here in Python, you have to escape the `\`, otherwise it just vanishes (just copy the string here into a python interactive shell and print it again, the backslash is gone, because it's interpreted).
So the final string here should be:
```
'+"A single quoted string with \\" in it"\n'
```
This produces then valid code when e.g. used for python strings. But this patch right now would produce invalid code, python would probably say something like "`it` is not defined" (because it sees `"A single quoted string with " in it ...`)

Reply:
Ohk so can you help me how to approach.